Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -325,8 +342,7 @@ void RenderGrid::layoutBlock(bool relayoutChildren, LayoutUnit)
 
         // Once grid's indefinite height is resolved, we can compute the
         // available free space for Content Alignment.
-        if (!hasDefiniteLogicalHeight)
-            m_trackSizingAlgorithm.setFreeSpace(ForRows, logicalHeight() - trackBasedLogicalHeight);
+        m_trackSizingAlgorithm.setFreeSpace(ForRows, logicalHeight() - trackBasedLogicalHeight);
```

Comment:
Rather than removing the check for `!hasDefiniteLogicalHeight`, can you just add `areMasonryRows` to it instead?

Seems less likely to break existing grid code that way.

Reply:
Fixed